hathor/transaction/base_transaction.py

```diff
@@ -424,8 +425,81 @@ def get_struct(self) -> bytes:
         struct_bytes += self.get_struct_nonce()
         return struct_bytes
 
+    def get_all_dependencies(self) -> Set[bytes]:
+        """Set of all tx-hashes needed to fully validate this tx, including parent blocks/txs and inputs."""
+        from hathor.transaction.genesis import GENESIS_HASHES
+        return set(chain(self.parents, (i.tx_id for i in self.inputs))) - set(GENESIS_HASHES)
+
+    def get_tx_dependencies(self) -> Set[bytes]:
+        """Set of all tx-hashes needed to fully validate this, except for block parent, i.e. only tx parents/inputs."""
+        from hathor.transaction.genesis import GENESIS_HASHES
+        parents = self.parents[1:] if self.is_block else self.parents
+        return set(chain(parents, (i.tx_id for i in self.inputs))) - set(GENESIS_HASHES)
+
+    def get_tx_parents(self) -> Set[bytes]:
+        """Set of parent tx hashes, typically used for syncing transactions."""
+        from hathor.transaction.genesis import GENESIS_HASHES
+        return set(self.parents[1:] if self.is_block else self.parents) - set(GENESIS_HASHES)
```

Comment:
I'll look into it on the other branches. It's something like where these are being used the genesis would have to be removed, so to remove this from all callers I pulled it into these methods. But I want to confirm this and it might be possible the callers won't have a problem of being returned a genesis tx, or would be better to handle it themselves.